### PR TITLE
Array list

### PR DESCRIPTION
**Pull request by V1talie**

### CODE REVIEW

**Comment by GAlexandru-be** [comment]
It will be really good to make your implementation parameterized

File: src/main/java/com/endava/internship/collections/StudentList.java [hunk 1]
```diff
@@ -1,40 +1,53 @@
 package com.endava.internship.collections;
 
-import java.util.Collection;
-import java.util.Iterator;
-import java.util.List;
-import java.util.ListIterator;
+import java.util.*;
```
Comment (GAlexandru-be) [inline src/main/java/com/endava/internship/collections/StudentList.java]: it is good for code readability to not use wildcard imports

Comment (V1talie) [reply src/main/java/com/endava/internship/collections/StudentList.java]: Done

File: src/main/java/com/endava/internship/collections/StudentList.java [hunk 2]
```diff
@@ -1,122 +1,205 @@
...
+            if (temp[i].equals(o)) occ = i;
+        }
+        return occ;
     }
 
     @Override
     public ListIterator<Student> listIterator() {
-        //TODO
-        return null;
+        return new MyIterator<>();
```
Comment (GAlexandru-be) [inline src/main/java/com/endava/internship/collections/StudentList.java]: remove generic brackets

File: src/main/java/com/endava/internship/collections/StudentList.java [hunk 3]
```diff
@@ -1,122 +1,205 @@
...
     }
 
     @Override
     public ListIterator<Student> listIterator(int i) {
-        //TODO
-        return null;
+        return new MyIterator<>(i);
```
Comment (GAlexandru-be) [inline src/main/java/com/endava/internship/collections/StudentList.java]: remove generic brackets

File: src/main/java/com/endava/internship/collections/StudentList.java [hunk 4]
```diff
@@ -142,4 +225,64 @@ public boolean retainAll(Collection<?> collection) {
         //Ignore this for homework
         throw new UnsupportedOperationException();
     }
+
+    private class MyIterator<Student> implements ListIterator<com.endava.internship.collections.Student> {
```
Comment (GAlexandru-be) [inline src/main/java/com/endava/internship/collections/StudentList.java]: remove the strange generic type for listiterator and replace with Student 
Also remove generic type for MyIterator

File: src/main/java/com/endava/internship/collections/StudentList.java [hunk 5]
```diff
@@ -142,4 +225,64 @@ public boolean retainAll(Collection<?> collection) {
...
+        }
+
+        @Override
+        public boolean hasPrevious() {
+            if(temp[current_pos-1]!= null) return true;
+            return false;
+        }
+
+        @Override
+        public com.endava.internship.collections.Student previous() {
```
Comment (GAlexandru-be) [inline src/main/java/com/endava/internship/collections/StudentList.java]: same as above

File: src/main/java/com/endava/internship/collections/StudentList.java [hunk 6]
```diff
@@ -142,4 +225,64 @@ public boolean retainAll(Collection<?> collection) {
...
+            return current_pos--;
+        }
+
+        @Override
+        public void remove() {
+            StudentList.this.remove(current_pos);
+        }
+
+        @Override
+        public void set(com.endava.internship.collections.Student student) {
```
Comment (GAlexandru-be) [inline src/main/java/com/endava/internship/collections/StudentList.java]: same as above

File: src/main/java/com/endava/internship/collections/StudentList.java [hunk 7]
```diff
@@ -142,4 +225,64 @@ public boolean retainAll(Collection<?> collection) {
...
+            StudentList.this.set(current_pos,student);
+        }
+
+        @Override
+        public void add(com.endava.internship.collections.Student student) {
```
Comment (GAlexandru-be) [inline src/main/java/com/endava/internship/collections/StudentList.java]: same as above

File: src/main/java/com/endava/internship/collections/StudentList.java [hunk 8]
```diff
@@ -142,4 +225,64 @@ public boolean retainAll(Collection<?> collection) {
...
+        }
+
+        @Override
+        public boolean hasNext() {
+            if(current_pos < size) return true;
+            return false;
+        }
+
+        @Override
+        public com.endava.internship.collections.Student next() {
```
Comment (GAlexandru-be) [inline src/main/java/com/endava/internship/collections/StudentList.java]: same as above

**Comment by GAlexandru-be** [comment]
ctrl+alt+l plz

File: src/main/java/com/endava/internship/collections/StudentList.java [hunk 9]
```diff
@@ -142,4 +225,64 @@ public boolean retainAll(Collection<?> collection) {
         //Ignore this for homework
         throw new UnsupportedOperationException();
     }
+
+    private class MyIterator<Student> implements ListIterator<com.endava.internship.collections.Student> {
+
+        int current_pos;
```
Comment (GAlexandru-be) [inline src/main/java/com/endava/internship/collections/StudentList.java]: use camel case plz

File: src/main/java/com/endava/internship/collections/StudentList.java [hunk 10]
```diff
@@ -142,4 +225,64 @@ public boolean retainAll(Collection<?> collection) {
...
+            current_pos = 0;
+        }
+
+        MyIterator(int index){
+            current_pos = index;
+        }
+
+        @Override
+        public boolean hasNext() {
+            if(current_pos < size) return true;
```
Comment (GAlexandru-be) [inline src/main/java/com/endava/internship/collections/StudentList.java]: 'if' statement can be simplified

File: src/main/java/com/endava/internship/collections/StudentList.java (hunk 2 above)
Comment (V1talie) [reply src/main/java/com/endava/internship/collections/StudentList.java]: Done

File: src/main/java/com/endava/internship/collections/StudentList.java (hunk 3 above)
Comment (V1talie) [reply src/main/java/com/endava/internship/collections/StudentList.java]: Done

File: src/main/java/com/endava/internship/collections/StudentList.java (hunk 4 above)
Comment (V1talie) [reply src/main/java/com/endava/internship/collections/StudentList.java]: Done

File: src/main/java/com/endava/internship/collections/StudentList.java (hunk 5 above)
Comment (V1talie) [reply src/main/java/com/endava/internship/collections/StudentList.java]: Done

File: src/main/java/com/endava/internship/collections/StudentList.java (hunk 6 above)
Comment (V1talie) [reply src/main/java/com/endava/internship/collections/StudentList.java]: Done

File: src/main/java/com/endava/internship/collections/StudentList.java (hunk 7 above)
Comment (V1talie) [reply src/main/java/com/endava/internship/collections/StudentList.java]: Done

File: src/main/java/com/endava/internship/collections/StudentList.java (hunk 8 above)
Comment (V1talie) [reply src/main/java/com/endava/internship/collections/StudentList.java]: Done

File: src/main/java/com/endava/internship/collections/StudentList.java (hunk 10 above)
Comment (V1talie) [reply src/main/java/com/endava/internship/collections/StudentList.java]: Done

File: src/main/java/com/endava/internship/collections/StudentList.java (hunk 9 above)
Comment (V1talie) [reply src/main/java/com/endava/internship/collections/StudentList.java]: Done

File: src/main/java/com/endava/internship/collections/StudentList.java [hunk 11]
```diff
@@ -142,4 +230,62 @@ public boolean retainAll(Collection<?> collection) {
         //Ignore this for homework
         throw new UnsupportedOperationException();
     }
+
+    private class MyIterator implements ListIterator<Student> {
+
+        int currentPosition;
```
Comment (GAlexandru-be) [inline src/main/java/com/endava/internship/collections/StudentList.java]: add private modifier

**Review by GAlexandru-be — APPROVED** [review]

File: src/main/java/com/endava/internship/collections/StudentList.java (hunk 11 above)
Comment (V1talie) [reply src/main/java/com/endava/internship/collections/StudentList.java]: Done

**Pull request merged**